mpop_service/facade.cpp

```diff
@@ -64,35 +65,31 @@ int Facade::getUserForTag(const QString& rfidTag) {
     }
     if (! tagExists) {
         visitorId = this->createTagAndUser(rfidTag);
-    } else {
-        // should not happen
     }
     return visitorId;
 }
 
 
 int Facade::createTagAndUser(const QString& rfidTag) {
     qDebug() << "createTagAndUser";
-    // return the created user id
-    // FIXME: It seems that createNewUser() should not be called every time. Check the logic.
-    int visitorId = createNewUser();
-    QString createSql = "INSERT INTO tag (`rfid`, `visitor_id`) VALUES (?, ?)";
-    QSqlQuery query(createSql, _database);
-    query.addBindValue(QVariant(rfidTag));
-    query.addBindValue(QVariant(visitorId));
-    query.exec();
-    bool ok = query.exec();
-    if (! ok) {
-        qWarning() << "ERROR: " << query.lastError().text();
-    }
+      int  visitorId = createNewUser();
+       QString createSql = "INSERT INTO tag (`rfid`, `visitor_id`) VALUES (?, ?)";
+       QSqlQuery query;
+       query.prepare(createSql);
+       query.addBindValue(QVariant(rfidTag));
+       query.addBindValue(QVariant(visitorId));
+       bool ok = query.exec();
+       if (! ok) {
+           qWarning() << "ERROR: " << query.lastError().text();
+       }
```

Comment:
I had fixed the indentation for that part. Let's not go back to a non-standard indentation. Qt has a very strict coding style that require us to indent with 4 spaces. You can use the auto-indent feature of Qt Creator to auto-format it.

Reply:
Ok sir